ext/curl/curl_file.c

```diff
@@ -117,6 +156,22 @@ ZEND_METHOD(CURLFile, getPostFilename)
 }
 /* }}} */
 
+/* {{{ proto void CURLFile::setFilename(string $name)
+   Set file name */
+ZEND_METHOD(CURLFile, setFilename)
+{
+	curlfile_set_property("name", INTERNAL_FUNCTION_PARAM_PASSTHRU);
+}
+/* }}} */
+
+/* {{{ proto void CURLFile::setBuffer(string $buffer)
+   Set buffer */
+ZEND_METHOD(CURLFile, setBuffer)
+{
+	curlfile_set_property("buffer", INTERNAL_FUNCTION_PARAM_PASSTHRU);
+}
+/* }}} */
```

Comment:
I think if filename always wins then maybe setting buffer while filename is already set should produce an exception. Though then it is not clear why these methods are needed at all - can't you just construct the proper one via factory method? Why one would need to change it on existing object? I'd much rather prefer it to be immutable.